### PR TITLE
Implement Sourcify to VerA script

### PR DESCRIPTION
**Pull request by marcocastignoli**
This script is needed to push all verified_contracts from Sourcify to VerA database. VerA is a live database in which other entities are writing while the script is running.

0. CURRENT_VERIFIED_CONTRACT is a variable stored in a permanent file, default 1.
1. Extract N verified_contracts starting from CURRENT_VERIFIED_CONTRACT
2. For each verified_contract:
   a. Update the CURRENT_VERIFIED_CONTRACT counter
   b. Get all information from verified_contracts
   c. Get all the FK information
   d. Insert all the verified_contracts dependencies (compiled_contracts,contract_deployments,contracts,code) and verified_contracts using the new ids as FKs

### CODE REVIEW

File: services/verifier-alliance/main.mjs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      `
+            SELECT * FROM verified_contracts
+            WHERE id >= $1
+            ORDER BY id ASC
+            LIMIT $2
+        `,
+      [CURRENT_VERIFIED_CONTRACT, N]
+    );
+
+    for (const contract of verifiedContracts) {
```
Comment (manuelwedler) [inline services/verifier-alliance/main.mjs:72]: So this script only process N contracts when it is run. This means the script should be run repeatedly to transfer all missing contracts. When would you stop running it? Why not iterate within the script until a stopping condition is reached?

File: services/verifier-alliance/main.mjs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+      const {
+        id: verifiedContractId,
+        deployment_id,
+        compilation_id,
+      } = contract;
+
+      // Update the CURRENT_VERIFIED_CONTRACT counter
+      CURRENT_VERIFIED_CONTRACT = verifiedContractId;
+      fs.writeFileSync(
```
Comment (manuelwedler) [inline services/verifier-alliance/main.mjs:81]: Wouldn't it be safer to only update this counter once we are sure the contract was written to the VerA database? If something goes wrong later, we might skip a contract, no?

File: services/verifier-alliance/main.mjs (hunk 1 above)
Comment (marcocastignoli) [reply services/verifier-alliance/main.mjs:72]: Uh, I forgot that part 🤣 I focused too much on the internal part that I forgot to conclude the loop

File: services/verifier-alliance/main.mjs (hunk 2 above)
Comment (marcocastignoli) [reply services/verifier-alliance/main.mjs:81]: the code is 

```
            WHERE id >= $1
            ORDER BY id ASC
            LIMIT $2
        `,
      [CURRENT_VERIFIED_CONTRACT, N]
```

so it starts from the last CURRENT_VERIFIED_CONTRACT. If the scripts breaks while synching a verified_contract then it starts again from that contract.

Comment (manuelwedler) [reply services/verifier-alliance/main.mjs:81]: Ah, I misunderstood it. Looks fine then :)

**Review by manuelwedler — CHANGES_REQUESTED** [review]
There is the missing loop, but other than that looks good to me

File: services/verifier-alliance/main.mjs (hunk 1 above)
Comment (marcocastignoli) [reply services/verifier-alliance/main.mjs]: Ok done @manuelwedler

**Review by manuelwedler — APPROVED** [review]
Looks good!

**Pull request merged**